### PR TITLE
Feature/uncertain paired data

### PR DESCRIPTION
**Pull request by Brennan1994**
confirmed toUncertainPairedData is working - inlcuding test.

### CODE REVIEW

File: ModelTests/IFdaFunctionShould.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            Statistics.IDistribution[] actualYFromUncertainPD = upd.ys();
+
+            for(int i = 0; i< func.Coordinates.Count; i++)
+            {
+                expectedXFromIFdaFunction[i] = func.Coordinates[i].X.Value();
+                var temp = (IDistributedOrdinate)func.Coordinates[i].Y;
+                expectedYFromIFdaFunciton[i] = temp.Dist;      
+            }
+
+            Assert.Equal(expectedXFromIFdaFunction, actualXFromUncertainPD);
```
Comment (HenryGeorgist) [inline ModelTests/IFdaFunctionShould.cs:42]: split this into two asserts. one comparing the fda function values to the listx and one comparing the uncertainpaired data to listx

that way we know our input is tracked through both objects.

File: Model/Functions/IFdaFunctionUtilities.cs [hunk 2]
```diff
@@ -27,7 +27,7 @@ public static class IFdaFunctionUtilities
         /// <param name="abbreviate"> <see langword="true"/> if an abbreviated label should be printed, <see langword="false"/> otherwise. </param>
         /// <returns> A <see cref="string"/> label. </returns>
         public static string PrintYLabel(IFdaFunction fx, bool abbreviate) => fx.XSeries.ParameterType.PrintLabel(fx.XSeries.Units, abbreviate);
-        public static object ToUncertainPairedData(this IFdaFunction fx)
+        public static UncertainPairedData ToUncertainPairedData(this IFdaFunction fx)
```
Comment (HenryGeorgist) [inline Model/Functions/IFdaFunctionUtilities.cs:30]: thanks

**Review by HenryGeorgist — CHANGES_REQUESTED** [review]

File: ModelTests/IFdaFunctionShould.cs (hunk 1 above)
Comment (Brennan1994) [reply ModelTests/IFdaFunctionShould.cs]: done

**Review by HenryGeorgist — APPROVED** [review]

File: ModelTests/IFdaFunctionShould.cs [hunk 3]
```diff
@@ -27,20 +28,21 @@ void ReturnEquivalentXandYasUncertainPairedData()
...
             {
                 expectedXFromIFdaFunction[i] = func.Coordinates[i].X.Value();
                 var temp = (IDistributedOrdinate)func.Coordinates[i].Y;
                 expectedYFromIFdaFunciton[i] = temp.Dist;      
             }
-
-            Assert.Equal(expectedXFromIFdaFunction, actualXFromUncertainPD);
-            Assert.Equal(expectedYFromIFdaFunciton, actualYFromUncertainPD);
+            Assert.Equal(expectedXFromUncertainPD, listx.ToArray());
+            Assert.Equal(expectedXFromIFdaFunction, listx.ToArray());
```
Comment (HenryGeorgist) [inline ModelTests/IFdaFunctionShould.cs:43]: great!

**Pull request merged**